kmod/src/block.c

```diff
@@ -1090,24 +1101,20 @@ u64 scoutfs_block_writer_dirty_bytes(struct super_block *sb,
  * atomically remove blocks when the only references are ours and the
  * hash table.
  */
-static int block_shrink(struct shrinker *shrink, struct shrink_control *sc)
+static unsigned long block_scan_objects(struct shrinker *shrink, struct shrink_control *sc)
 {
-	struct block_info *binf = container_of(shrink, struct block_info,
-					       shrinker);
+	struct block_info *binf = container_of(shrink, struct block_info, shrinker);
 	struct super_block *sb = binf->sb;
 	struct rhashtable_iter iter;
 	struct block_private *bp;
 	bool stop = false;
+	unsigned long freed = 0;
 	unsigned long nr;
 	u64 recently;
 
-	nr = sc->nr_to_scan;
-	if (nr == 0)
-		goto out;
-
-	scoutfs_inc_counter(sb, block_cache_shrink);
+	scoutfs_inc_counter(sb, block_cache_scan_objects);
 
-	nr = DIV_ROUND_UP(nr, SCOUTFS_BLOCK_LG_PAGES_PER);
+	nr = DIV_ROUND_UP(sc->nr_to_scan, SCOUTFS_BLOCK_LG_PAGES_PER);
```

Comment:
The changes don't maintain this scaling from object (block) counts to scaled by their number of pages.  It scales to nr on input and then returns freed unscaled.  Let's just remove all this scaling entirely.  All the _count functions can stop doing it, and all the inputs of the scan functions can stop doing it.  I don't think it ever amounted to everything and it's messing us up.